CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ endif()
 include(CheckCCompilerFlag)
 
 if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(MACOS TRUE)
```

Comment:
Up to now we check `__APPLE__` in the C source (and `APPLE` in some CMakeLists.txt files). Is there a reason to prefer one or the other?

Reply:
> Is there a reason to prefer CMake's CMAKE_SYSTEM_NAME check?

Not that I know of.  I can switch to `__APPLE__`.

Reply:
> Is there a reason to prefer CMake's CMAKE_SYSTEM_NAME check?

`CMAKE_SYSTEM_NAME` is the system you're building for.  If I understand correctly, `UNIX`/`APPLE`/`WIN32` is for the system `cmake` is running on, which can vary for things like cross-compiling.

Also, `UNIX` is true for Apple and Cygwin.